infocmdb/workflow.go

```diff
@@ -132,3 +135,22 @@ func (iw *IntWrapper) UnmarshalJSON(b []byte) error {
 func (c *Client) GetWorkflowContext(workflowInstanceId int) (workflowContext *v2.WorkflowContext, err error) {
 	return c.v2.GetWorkflowContext(workflowInstanceId)
 }
+
+func (c *Client) GetWorkflowUserId() (userid int, err error) {
```

Comment:
This function reparses the process arguments every time it is called.
This is not good for performance, but it is even worse that some functions now (silently) require that those arguments are present.
Although most of the time this sdk is used inside workflows there are some situations were this is not the case.
If we need the workflow parameters (or parts of it) we should rather require them to be passed as function arguments.

Reply:
I just re-reviewed this part and your initial comment.
This function is only called in case the user id wasn't already provided, but I do still think that this behavior is not very intuitive and am also not happy with the reparsing of the process arguments here.

I'd prefer the user id argument to be required for the functions that need it. I think it is accessible enough in the workflow as they are already passed as a parameter in the workflow function itself.

Reply:
I added the user id to the workflow paramas.
It can be now accesed like this in the workflow function:

func main() {
	w := infocmdb.NewWorkflow()
	w.Run(workflow)
}

func workflow(params infocmdb.WorkflowParams, cmdb *infocmdb.Client) (err error) {

	log.Info(params.UserId)